minifi-c2/minifi-c2-commons/src/main/java/org/apache/nifi/minifi/c2/model/extension/BundleManifest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.c2.model.extension;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.List;
+
+@ApiModel
+public class BundleManifest {
+
```

Comment:
How would one define an extension outside of the confines of a manifest?

Reply:
::sigh:: I typed a long response that I think got lost before I hit the button.

The gist of it is -> I need to think about this. Currently the data model requires defining an extension in the context of a bundle. If there is no actual bundle, you could use a default bundle (I even included a constructor `defaultBundle()` to create one) that indicates there is no actual bundle used. 

Alternatively, we could add ExtensionComponents directly to AgentManifest so you could include them without creating fake bundles to act as containers. Thoughts?

Reply:
I think I like the idea of the direct access to add components.  Bundles are nice for the Java side of the house but don't necessarily map well to C++.  From the standpoint of the flow designer, the notions of bundles should be minimal but some defined tuple (class, version, etc) would be all that matters as a keyed extension (which has a fuller model as outlined with the other work that has gone on in this PR).

Reply:
OK, I like that too. I'll add top level components to the AgentManifest, and devs can choose which makes more sense for them. Flow Designer can adapt to both.